filter/fieldnamefinder.go

```diff
@@ -0,0 +1,110 @@
+package filter
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+type fieldNameFinder struct {
+	stringCaseFinder *stringCaseFinder
+}
+
+func newFieldNameFinder(pattern string) *fieldNameFinder {
+	return &fieldNameFinder{makeStringCaseFinder(pattern)}
+}
+
+// find returns true if buf may contain a record with a field whose
+// fully-qualified name (e.g., a.b.c) matches the pattern. find also returns
+// true if it encounters an error.
+func (f *fieldNameFinder) find(zctx *resolver.Context, buf []byte) bool {
+	for len(buf) > 0 {
+		if buf[0]&0x80 != 0 {
+			// Control messages are not expected.
+			return true
+		}
+		// Read uvarint7 encoding of type ID.
+		var id int
+		if buf[0]&0x40 == 0 {
+			id = int(buf[0])
+			buf = buf[1:]
+		} else {
+			v, n := binary.Uvarint(buf[1:])
+			if n <= 0 {
+				return true
+			}
+			id = int((v << 6) | uint64(buf[0]&0x3f))
+			buf = buf[1+n:]
+		}
+		length, n := binary.Uvarint(buf)
+		if n <= 0 {
+			return true
+		}
+		buf = buf[n+int(length):]
+		t, err := zctx.LookupType(id)
```

Comment:
I think you've mentioned the possibility of caching results for field name lookup. Could you cache the result for this type id, within this call to `find`, or at a higher level?

Reply:
ZNG end-of-stream markers together with the current context.Resolver interface mean it can only be done within a call to find, as in 9c5908311b162fc434734fabab16a539934bbd9b.